$OpenBSD: patch-src_libutil_util_c,v 1.4 2020/02/11 15:16:22 sthen Exp $

- libressl doesn't have FIPS mode, so FIPS_mode and friends don't exist

- openblas_set_bum_threads is only on OpenBLAS not plain BLAS

Index: src/libutil/util.c
--- src/libutil/util.c.orig
+++ src/libutil/util.c
@@ -2484,6 +2484,9 @@ rspamd_config_libs (struct rspamd_external_libs_ctx *c
 		}
 
 		if (cfg->fips_mode) {
+#ifdef LIBRESSL_VERSION_NUMBER
+			msg_info_config ("LibreSSL does not support FIPS mode");
+#else
 			int mode = FIPS_mode ();
 			unsigned long err = (unsigned long)-1;
 
@@ -2505,6 +2508,7 @@ rspamd_config_libs (struct rspamd_external_libs_ctx *c
 			else {
 				msg_info_config ("OpenSSL FIPS mode is enabled");
 			}
+#endif
 		}
 
 		if (cfg->ssl_ca_path) {
@@ -2553,7 +2557,7 @@ rspamd_config_libs (struct rspamd_external_libs_ctx *c
 			ZSTD_freeCStream (ctx->out_zstream);
 			ctx->out_zstream = NULL;
 		}
-#ifdef HAVE_CBLAS
+#if 0
 		openblas_set_num_threads (cfg->max_blas_threads);
 #endif
 	}
